### ISSUE TITLE
Rename `ITK` repository default branch from `master` to `main`

### ISSUE THREAD

**Issue by thewtex** [InsightSoftwareConsortium/ITK#4732]
### Description

`main` is now the default branch for GitHub projects as opposed to `master`. Provide the expected default.

### Impact analysis

This will provide the expected default branch name, and it will also cause disruption to existing clones and forks.

### Versions

Target for ITK 6.

### Additional Information

Todo:

- [x] Update related repositories
  - [x] Update remote module's default branches (in progress)
  - [x] Update ITKSphinxExamples
  - [x] Update ITKSoftwareGuide
  - [x] Update ITK-Wasm
  - [x] Update itk-viewer
  - [x] Update InsightSoftwareConsortium/ITKClangFormatLinterAction
  - [x] Update InsightSoftwareConsortium/ITKPythonPackage
- [x] Community notices (Discourse, etc.): https://discourse.itk.org/t/itk-git-repository-primary-branch-name-transition-from-master-to-main/7569
- [x] https://adamj.eu/tech/2022/10/27/git-how-to-alias-master-as-main/
- [x] Dashboard Scripts #5429 
- [x] Scripts in the repository #5426 
- [x] Release scripts #5426
- [x] Dashboard systems
- [x] Setup nightly sync of `master` backwards compatibility branch to `main`
- [x] Add to documentation on how to update local clone configuration and GitHub forks #5426 (SetupForDevelopment.sh does this)
- [x] GitHub Repository Settings Rename `master` default branch to `main` 

@hjmjohnson

**Comment by dzenanz** [comment]
Welcome to the ITK community Matt 😄

**Comment by jhlegarreta** [comment]
All remotes having a `LICENSE` badge will surely have the default branch name hard-coded, e.g.
https://github.com/InsightSoftwareConsortium/ITKIOTransformDCMTK/blob/master/LICENSE

https://github.com/InsightSoftwareConsortium/ITKIOTransformDCMTK/pull/14/files#diff-7b3ed02bc73dc06b7db906cf97aa91dec2b2eb21f2d92bc5caa761df5bbc168fR9

So after the default branch gets renamed to `main` these links will no longer work. Having an easy way to apply changes to all remotes with a single command for this kind of changes is becoming a must, otherwise maintaining them is very time-consuming.

**Comment by thewtex** [comment]
I have started to rename the remote repository branches to `main` in the process of #4748 . We can use the all-repos tool to update the README badges.

**Comment by jhlegarreta** [comment]
Have renamed the default branch in `ITKSoftwareGuide`.

**Comment by thewtex** [comment]
Some WIP in #5426

**Comment by thewtex** [comment]
@hjmjohnson since you will be at Project Week, are you still OK with going ahead with this on Monday?

**Comment by thewtex** [comment]
@seanm could your nightly dashboard scripts please be updated to follow `main`?

**Comment by seanm** [comment]
Ah, I see it's in `itk_common.cmake` ... I guess that file was updated by you guys?

**Comment by thewtex** [comment]
@seanm yes, `itk_common.cmake` has been updated on the ITK `dashboard` branch, and you could download that for new defaults, or set:

```
set(dashboard_git_branch main)
```

**Comment by jhlegarreta** [comment]
@thewtex should I go ahead and rename the branch in the Examples repository?
https://github.com/InsightSoftwareConsortium/ITKSphinxExamples/pull/442

BTW, when `InsightSoftwareConsortium/ITKClangFormatLinterAction` will get renamed all actions across all repositories will need to be renamed

**Comment by SimonRit** [comment]
> * https://github.com/RTKConsortium/RTK

I don't fully agree with the move but we'll comply and rename RTK, CudaCommon and PCT (soon to be a remote module). Is there a recipe for renaming the master branch of a module?

**Comment by dzenanz** [comment]
I just renamed
- https://github.com/KitwareMedical/ITKDICOM
- https://github.com/scifio/scifio-imageio

**Comment by dzenanz** [comment]
Looking at the [dashboard](https://open.cdash.org/index.php?project=Insight), it looks like all the machines are submitting correctly. I checked the appropriate box in the issue description.

**Comment by dzenanz** [comment]
> Is there a recipe for renaming the master branch of a module?

@SimonRit Rename it in settings, and update mentions of `master` to `main` in CI, and possibly other places (such as documentation).

**Comment by dzenanz** [comment]
See examples:
- https://github.com/KitwareMedical/ITKDICOM/pull/12
- https://github.com/scifio/scifio-imageio/pull/77
